src/tasks/language.py

```diff
@@ -221,3 +220,56 @@ def _save(self):
 
         )
         return False
+
+class Listening(Task):
+
+    DEFAULT_INSTRUCTION = """You are about to listen to audio stimuli"""
+
+    INSTRUCTION_WAIT_KEY = (
+        DEFAULT_INSTRUCTION + "\nWhen you're ready press <%s>" % MOVE_ON
+    )
+
+    def __init__(self, filepath, *args, **kwargs):
+        self.wait_key = kwargs.pop("wait_key", False)
+        super().__init__(**kwargs)
+        if self.wait_key:
+            self.instruction = Listening.INSTRUCTION_WAIT_KEY
+        self.filepath = filepath
+        if not os.path.exists(self.filepath):
+            raise ValueError("File %s does not exists" % self.filepath)
+
+    def _instructions(self, exp_win, ctl_win):
+        screen_text = visual.TextStim(
+            exp_win,
+            text=self.instruction,
+            alignText="center",
+            color="white",
+            wrapWidth=config.WRAP_WIDTH,
+        )
+
+        def _draw_instr():
+            screen_text.draw(exp_win)
+            if ctl_win:
+                screen_text.draw(ctl_win)
+
+        if self.wait_key:
+            while True:
+                if len(event.getKeys([MOVE_ON])):
+                    break
+                _draw_instr()
+                yield
+        else:
+            for frameN in range(config.FRAME_RATE * config.INSTRUCTION_DURATION):
+                _draw_instr()
+                yield
+
+    def _run(self, exp_win, ctl_win):
```

Comment:
It seems like the run here should present more than one stimuli. That function is supposed to host the task logic (timing, ISI, keypresses of responses, ....)